tests/test_pumptrap_calibration.py

```diff
@@ -0,0 +1,274 @@
+            
+            trap_dict[key] = {}
+            trap_dict[key]['cap']  = [trap_pump[4], 0, trap_pump[5]] #Add the error in to keep the expected dimensions
+            trap_dict[key]['E'] = trap_pump[6]
+            trap_dict[key]['cs'] = trap_pump[7]
+            trap_dict[key]['Rsq'] = trap_pump[8]
+            trap_dict[key]['tau at input T'] = trap_pump[9]
+
+        return trap_dict
+
+def test_tpump_analysis():
+    '''
+    After reading in the data and some setup, then run the subset of appropriate tests from
+    test_tfit_const_True_sub_noise_ill in ut_tpump_final.py
+    '''
+
+    # Set the seed - II&T ut tests don't work everytime, so let's fix it. 
+    np.random.seed(39)
+
+    #Generate the mock data:
+    test_data_dir = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'test_data', "pump_trap_data_test")
+    metadata_file = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'test_data', "metadata_test.yaml")
+    print("Generating mock data")
+    generate_mock_pump_trap_data(test_data_dir, metadata_file)
+    print("Done generating mock data")
+
+    #Read in all the data. 
+    # test_data_dir = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'test_data', "pump_trap_data")
+    data_filenames = sorted(glob.glob(os.path.join(test_data_dir, "*.fits")))
+    pump_trap_dataset = Dataset(data_filenames)
+
+    #Parse the first three characters of each filename into a temperature
+    # temps = [int(os.path.basename(f)[:3]) for f in data_filenames]
+    # npumps = [int(os.path.basename(f).split("_")[5]) for f in data_filenames]
+    # scheme = [int(os.path.basename(f).split("_")[2]) for f in data_filenames]
+    #Hack in some missing header parameters. 
+    arrtype = 'SCI'
+    # em_gain = 10 #The default in generate_test_data.py
+
+    for j,frame in enumerate(pump_trap_dataset):
+        frame.ext_hdr['ARRTYPE'] = arrtype
+        # frame.ext_hdr['CMDGAIN'] = em_gain
+        # frame.ext_hdr['EXCAMT'] = temps.pop(0)
+
+        #Get the scheme the the "scheme" list. For the current scheme set the header keyword TPSCHEM* 
+        # equal to the npumps for this filename, where * is equal to the scheme. For the other schemes (up to * =4) set TPSCHEM* =0.
+        # for i in range(1, 5):
+        #     if scheme[j] == i:
+        #         frame.ext_hdr['TPSCHEM' + str(i)] = npumps.pop(0)
+        #     else:
+        #         frame.ext_hdr['TPSCHEM' + str(i)] = 0
+
+        #Get the phase time from the filename: its between the string "phasetime" and the ".fits" extension at the end
+        # phase_time = os.path.basename(data_filenames[j]).split("phasetime")[1].split(".fits")[0]
+        # frame.ext_hdr['TPTAU'] = float(phase_time)
+
+    #Run the bias subtraction
+    # #TODO Figure out which detector regions to pass in here. 
+    # arrtype = pump_trap_dataset[0].ext_hdr['ARRTYPE']
+
+    #Detector regions for the smaller pump_trap_data - taken from metadata_test.yaml
+    detector_regions = {
```

Comment:
One other thing:  This is in mocks.py, so you could do `from corgidrp.mocks import detector_areas_test` to get this instead of copying it here.

Reply:
good call!

Reply:
good call!